File: VsVim/ICompletionWindowBroker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.Composition;
using Microsoft.VisualStudio.Text.Editor;

namespace VsVim
{
    /// <summary>
    /// Represents the Vim sense of windows which are involved in completion.  It essentially 
    /// aggregates the results of intellisense and signature info.  
    /// </summary>
    public interface ICompletionWindowBroker
    {
        bool IsCompletionWindowActive(ITextView view);
        void DismissCompletionWindow(ITextView view);
    }
}

File: VimCore/ICompletionWindowBroker.fs
#light

namespace Vim
open Microsoft.VisualStudio.Text.Editor

/// Used to determine if a completion window is active for a given view
type ICompletionWindowBroker =

    /// TextView this broker is associated with
    abstract TextView : ITextView 

    /// Is there currently a completion window active on the given ITextView
    abstract IsCompletionWindowActive : bool

    /// Dismiss any completion windows on the given ITextView
    abstract DismissCompletionWindow : unit -> unit
